### ISSUE TITLE
Importing “Bechdel Test” from prod loads everything as a string

### ISSUE THREAD

**Issue by snickell** [code-dot-org/code-dot-org#56771]
From Molly: I imported “Bechdel Test” from prod. Everything is turned into a string, and all integers become doubles (ex. 2013 -. “2013.0”). This works ok for the columns that are actually numbers, because I can click “convert to number”. However, for the column “title” most are strings, but one is 42, which turns into “42.0” and I would need to manually convert it. Tested in prod and the types don’t get converted to strings on import

**Comment by snickell** [comment]
@molly-moen do you remember how you imported "bechdel test" from prod? Was it a CSV export followed by a CSV import?

**Comment by molly-moen** [comment]
Yep I exported the csv from prod, then imported onto the adhoc

**Comment by snickell** [comment]
@cnbrenci we missed this snippet from `parseRecordsDataFromCsv()` (from firebaseStorage.js):
```
      for (const key in record) {
        record[key] = castValue(record[key], /* allowUnquotedStrings */ true);
      }
```

Which calls `castValue()`: https://github.com/code-dot-org/code-dot-org/blob/ae1eef1d6e024937c2f914e7b36bfd4e2f8feb56/apps/src/storage/dataBrowser/dataUtils.js/#L75-L106

In contrast, we ship the CSV text straight to the backend, and do the parsing on the Rails side.

I could see two approaches to fixing this... one would be to reimplement the `castValue()` in ruby, the other would be to actually re-use `parseRecordsDataFromCsv()` and `castValue()`, and parse the CSV in JS same as we do for firebase, and implement a `DatablockStorageController.create_records` call (which we'll need later for batching anyway, and we already have `table.create_records [record_json]` implemented, so the hard part is done).

**Comment by snickell** [comment]
I believe this should be fixed by #56902, pending verification on the Bechdel Test dataset.

**Comment by snickell** [comment]
Verify fix works on Bechdel Test.

On datablock storage:
![Image](https://github.com/code-dot-org/code-dot-org/assets/223277/641e0d4d-275d-4726-bab6-44623e8ee74b)

On prod:
![Image](https://github.com/code-dot-org/code-dot-org/assets/223277/5ba194be-f456-4cf1-bb86-db064cb51c5e)

Prod doesn't actually preserve types on import.... because the CSV is typeless. Instead, prod and now datablock storage auto-casts the strings on import_csv using JSON.parse on a per-value basis. As a result of this approach, prod and datablock storage share a "bug" where 42 is a number rather than a string.

Thanks for finding this @molly-moen 🙏
